test/extensions/tracers/zipkin/span_buffer_test.cc

```diff
@@ -10,13 +14,37 @@ namespace Tracers {
 namespace Zipkin {
 namespace {
 
+void expectValidBufferedProtoListOfSpans(const SpanBuffer& buffer) {
+  std::string expected_yaml = "spans:";
+  for (const auto& span : buffer.spans()) {
+    const std::string expected_span_yaml = fmt::format(
+        R"EOF(
+- traceId: {}
+  parentId: {}
+  id: {}
+)EOF",
+        Base64::encode(span.traceIdAsByteString().c_str(), span.traceIdAsByteString().size()),
+        span.isSetParentId() ? Base64::encode(span.parentIdAsByteString().c_str(),
+                                              span.parentIdAsByteString().size())
+                             : "",
```

Comment:
maybe trailing comment that empty is same as not serializing the field. We shouldn't serialize on empty